### PR TITLE
Update implementation documentation

### PR DESCRIPTION
**Pull request by WilcoFiers**
The purpose of this PR is to provide clear guidance about how to produce an implementation for ACT-R. It includes

- How to use test cases
- How to submit a report
- What the report format should look like
- How is the "correctness" of an implementation determined
- How checks in an implementation are mapped to ACT-R rules

Not included:

- How to use the WCAG-EM Report tool for implementations

## Guidance for the PR (pull request) creator

**When creating PR:**
- [ ] Make sure you requesting to **pull a issue/feature/bugfix branch** (right side) to the master branch (left side).

**After creating PR:**
- [ ] Add yourself (and co-authors) as "Assignees" for PR
- [ ] Add label to indicate if it's a `Rule`, `Definition` or `Chore` (more to the administrative side)
- [ ] Add relevant project (e.g. "Q3 2018 Status") to PR
- [ ] OPTIONAL: If you want anyone in particular to review your pull request, assign them as "Reviewers".
- [ ] Close the issue that the PR resolves (and make sure the issue is referenced in the top of this comment)

# How to Review And Approve
- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines. 
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.

### CODE REVIEW

File: pages/implementations/mapping.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+The correctness of an implementation is based on the results from running the [test cases](../testcases/) in a tool or test procedure. The three types of test cases have a range of outcomes that are allowed:
+
+| Test Case Type | Allowed outcomes                                   |
+| -------------- | -------------------------------------------------- |
+| Passed         | `passed`, `cantTell` or `inapplicable`             |
+| Failed         | `failed` or `cantTell`                             |
+| Inapplicable   | `inapplicable`, `cantTell`, `passed` or `untested` |
+
+A check in an implementation that returns one of the allowed outcomes for all test cases in an ACT-R rule is a _correct implementation_. If the results don't include `cantTell`, that implementation is considered an _automated implementation_.
```
Comment (kasperisager) [inline pages/implementations/mapping.md]: > If the results don't include `cantTell`, that implementation is considered an _automated implementation_.

This is only true if we assume that the implementation was executed without human input, which is for example not how we acceptance test Alfa. See: https://github.com/siteimprove/alfa-act-r

File: pages/implementations/mapping.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+An implemented check that has all passed and inapplicable test cases correct, but only has some of the failed test cases correct is called a _partial implementation_. If there are partial implementations for an ACT-R rule that together get all failed test cases correct, this _set of checks_ is considered a _correct implementation_.
+
+**Note**: The word "check" is used to mean the implementation of a rule.
```
Comment (kasperisager) [inline pages/implementations/mapping.md]: I question the usefulness of introducing a new term that also means "implementation". Seems like "implementation" is sufficient.

File: pages/implementations/reporting.md [hunk 3]
```diff
@@ -1,99 +1,17 @@
...
+1. [Join the ACT-R Community group](https://w3.org/community/act-r/). This ensures the community license agreement is signed.
 
-Test tool authors are welcome to submit the implementation results for the above test cases to the ACT-Rules Community. The implementation report should be a valid [JSON-LD](https://json-ld.org/spec/latest/json-ld/) document, where the assertions are expressed using [EARL](https://www.w3.org/TR/EARL10-Schema/).
+2. [Open an issue](https://github.com/act-rules/act-rules.github.io/issues/new), proposing the new implementation. Please include the name of the implementation, and a URL to the EARL report.
 
-Also, note that the report contains a list of assertion results for each of the test cases (of each rule). An example assertion for a test case, is as below:
+> Note: Refer [available tools](/pages/implementations/tools/) section, to see if any of them can be used to generate the report in the EARL format.
```
Comment (kasperisager) [inline pages/implementations/reporting.md]: "Refer _to the_"?

File: pages/implementations/testcases.md [hunk 4]
```diff
@@ -2,28 +2,56 @@
...
- ]
+      "rulePage": "https://act-rules.github.io/rules/5f99a7",
+      "requirementsMapping": ["wcag20:4.1.2"]
+    }, ... ]
+}
 ```
+
+## Running The Test Cases
+
+To run the test cases, go through each URL run the necessary tests on that page. Tools and test procedures do not need to have a one-to-one mapping to ACT-R rules. While you could run all tests in an implementation on every test case, for the sake of efficiency, we recommend doing one of the following:
```
Comment (kasperisager) [inline pages/implementations/testcases.md]: "_and_ run"?

Comment (jkodu) [inline pages/implementations/testcases.md]: ... `one-to-one mapping to ACT Rules` ...

ACT-R rules does not read right...

File: pages/implementations/testcases.md [hunk 5]
```diff
@@ -2,28 +2,56 @@
...
+
+Correctness of an implementation is based on the results for test cases. See [implementation mapping](../mapping/) for more information.
+
+## Contribute An Implementation
+
+If you developed an accessibility tool or a testing methodology, and would like to have your implementation included in the ACT-R website, there are two ways you can do so.
+
+1. If you have a tool that can return a data format, you will need to run your tests against the ACT-R test cases and [submit a report](../reporting/).
+
+2. If you use manual test methodology, where you fill results into some report template or tool, you can [Use the WCAG-EM Report Tool](../wcag-em-tool/) instead to produce implementation reports.
```
Comment (jkodu) [inline pages/implementations/testcases.md]: Step 2 is not clear that on how to submit the report after using the WCAG-EM Report Tool to produce the report.

File: pages/implementations/reporting.md [hunk 6]
```diff
@@ -1,99 +1,17 @@
...
-| Status        | Description                                                                 |
-| ------------- | --------------------------------------------------------------------------- |
-| `New`         | The rule has no implementation                                              |
-| `In Progress` | The rule has at least one implementation but no more than 2 implementations |
-| `Done`        | The rule has 3 or more implementations                                      |
-
-The higher the number of implementations, collected by the various community submitted reports, ensure high accuracy and completeness of the rules.
-
-> Note: Refer [available tools](/pages/implementations/tools) section, to see if any of them can be used to generate the report in the above format.
+Submitting an implementation requires permission from the owner of the test tool / methodology that is proposed. If you wish to submit an implementation on behalf of an implementor, we will require written permission from the owner of the implementation sent to the ACT-R public mailinglist.
```
Comment (jkodu) [inline pages/implementations/reporting.md]: `mailinglist` > `mailing list`

File: pages/implementations/overview.md [hunk 7]
```diff
@@ -1,13 +1,25 @@
...
+| `In Progress` | The rule has less than 3 implementations |
+| `Done`        | The rule has 3 or more implementations   |
+
+## Contribute An Implementation
+
+If you developed an accessibility tool or a testing methodology, and would like to have your implementation included in the ACT-R website, there are two ways you can do so.
+
+1. If you have a tool that can return a data format, you will need to run your tests against the [ACT-R test cases](../testcases/) and [submit a report](../reporting/).
+
+2. If you use manual test methodology, where you fill results into some report template or tool, you can [Use the WCAG-EM Report Tool](../wcag-em-tool/) instead to produce implementation reports.
```
Comment (jkodu) [inline pages/implementations/overview.md]: Step 2 is not clear that on how to submit the report after using the WCAG-EM Report Tool to produce the report.

File: pages/implementations/mapping.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
+---
+title: Mapping To Rule
+---
+
+## Implementation Scoping
+
+An implementation does not need to have a one-to-one mapping to ACT-R rules. An implementation can test multiple ACT-R rules in a single check. In manual test procedures, it is common to test multiple things at once. For instance, a test procedure looking at the language of a page will simultaniously test whether or not the `<html>` element has a `lang` attribute, if the value of the `lang` attribute is valid, and if the value matches the language of the page. This is 3 separate rules, covered in a single test procedure of an implementation.
```
Comment (jkodu) [inline pages/implementations/mapping.md]: `ACT-R rules` > `ACT Rules`

Comment (WilcoFiers) [reply pages/implementations/mapping.md]: Disagree. ACT-R rules are rules written by the ACT-R community group. ACT rules are rules conforming to the ACT Rules Format.

File: pages/implementations/overview.md (hunk 7 above)
Comment (WilcoFiers) [reply pages/implementations/overview.md]: I'll fill out that page later.

File: pages/implementations/testcases.md (hunk 4 above)
Comment (WilcoFiers) [reply pages/implementations/testcases.md]: Rules from the ACT-R community group are ACT-R rules.

File: pages/implementations/testcases.md (hunk 5 above)
Comment (WilcoFiers) [reply pages/implementations/testcases.md]: I'll add the WCAG-EM stuff later.

**Pull request merged**